library/std/src/sys/pal/unix/process/process_common.rs

```diff
@@ -426,9 +442,25 @@ impl CStringArray {
     }
 }
 
-fn construct_envp(env: BTreeMap<OsString, OsString>, saw_nul: &mut bool) -> CStringArray {
+fn construct_envp(
+    env: BTreeMap<OsString, OsString>,
+    saw_nul: &mut bool,
+    saw_invalid_env_key: &mut bool,
+) -> CStringArray {
     let mut result = CStringArray::with_capacity(env.len());
     for (mut k, v) in env {
+        {
+            let mut iter = k.as_bytes().iter();
+            if iter.next().is_none() {
+                *saw_invalid_env_key = true;
+                continue;
+            }
+            if iter.any(|&b| b == b'=') {
+                *saw_invalid_env_key = true;
+                continue;
+            }
+        }
```

Comment:
I think this would be simpler without an iterator:
```suggestion
        if k.is_empty() || k.as_bytes()[1..].contains(&b'=') {
            *saw_invalid_env_key = true;
            continue;
        }
```

Reply:
I guess it's still obvious enough that there's no panic.

Reply:
Done.